### PR TITLE
Make payload size configurable

### PR DESCRIPTION
**Pull request by maknahar**

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -16,6 +16,25 @@ import (
...
+func init() {
+	downloadConcurrency, _ = strconv.Atoi(os.Getenv("DOWNLOAD_CONCURRENCY"))
+	if downloadConcurrency <= 0 {
+		downloadConcurrency = 10
+		log.Println("DOWNLOAD_CONCURRENCY not specified. Setting it to", downloadConcurrency)
+	}
+
+	maxPayloadSizeInByte, _ = strconv.ParseInt(os.Getenv("MAX_PAYLOAD_SIZE_IN_BYTES"), 10, 64)
+	if maxPayloadSizeInByte <= 0 {
+		maxPayloadSizeInByte = 32 * 1024 * 1024 * 1024
```
Comment (thamizhv) [inline main.go]: How about 32 << 30?

Comment (arjunmayilvaganan) [inline main.go]: @maknahar 
I think this is 32GB. Correct me If I'm wrong.

File: main.go [hunk 2]
```diff
@@ -16,6 +16,25 @@ import (
 	_ "github.com/heroku/x/hmetrics/onload"
 )
 
+var (
+	downloadConcurrency  int
+	maxPayloadSizeInByte int64
+)
+
+func init() {
+	downloadConcurrency, _ = strconv.Atoi(os.Getenv("DOWNLOAD_CONCURRENCY"))
```
Comment (arjunmayilvaganan) [inline main.go:25]: I recommend splitting this step into two.
Can we continue with default value if `os.Getenv("DOWNLOAD_CONCURRENCY")` is `""`, i.e., unset,
and
abort if we find unexpected value in `DOWNLOAD_CONCURRENCY` (i.e., `err != nil` for `strconv.Atoi()`)?

File: main.go (hunk 1 above)
Comment (arjunmayilvaganan) [reply main.go]: I think expressing like it is currently in the change is more readable.

File: main.go [hunk 3]
```diff
@@ -102,19 +127,34 @@ func handleError(err error, w http.ResponseWriter) {
 func zipper(writer io.Writer, incoming chan RemoteFile, complete chan error) {
 	archive := zip.NewWriter(writer)
 	for rf := range incoming {
+
 		zipEntryHeader := &zip.FileHeader{
 			Name:   rf.path,
 			Method: zip.Deflate,
 		}
+
 		zipEntryHeader.SetModTime(time.Now())
```
Comment (thamizhv) [inline main.go]: SetModTime is deprecated it seems. can use as below
```
zipEntryHeader.SetModTime(time.Now())
zipEntryHeader.Modified = time.Now().UTC()
```

File: main.go (hunk 1 above)
Comment (maknahar) [reply main.go]: @thamizhv nice way to do that if we are writing performance critical software. Would have more performant to calculate manually and assign a number. Point is to make it more readable and easy to change.

Comment (maknahar) [reply main.go]: 1024 byte * 1204 * 1024 * 32 = 32 MB. No?

File: main.go (hunk 2 above)
Comment (maknahar) [reply main.go:25]: Could you explain the benefits?

File: main.go (hunk 3 above)
Comment (maknahar) [reply main.go]: Nice catch. Changing it.

File: main.go (hunk 1 above)
Comment (thamizhv) [reply main.go]: No... It is 32 GB...

Comment (maknahar) [reply main.go]: You guys are correct. Changing it.

File: main.go (hunk 2 above)
Comment (arjunmayilvaganan) [reply main.go:25]: Just a nice to have.

1. If the value is not set, we'll be going with the default value.
2. If the value set, but is invalid, shouldn't we know about the issue?

Maybe we need not abort it, but maybe we could just log it?

Comment (maknahar) [reply main.go:25]: @arjunmayilvaganan, In either case, we need to check and ser the value. The default value is there just as a backup.

**Review by arjunmayilvaganan — APPROVED** [review]

**Review by thamizhv — APPROVED** [review]

**Pull request merged**